app/src/main/java/com/bashkevich/androidfundamentals/moviesdetails/view/FragmentMoviesDetails.kt

```diff
@@ -0,0 +1,113 @@
+import android.widget.RatingBar
+import android.widget.TextView
+import androidx.fragment.app.viewModels
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.bashkevich.androidfundamentals.R
+import com.bashkevich.androidfundamentals.data.Movie
+import com.bashkevich.androidfundamentals.moviesdetails.viewmodel.MoviesDetailsViewModel
+import com.bashkevich.androidfundamentals.moviesdetails.viewmodel.MoviesDetailsViewModelFactory
+import com.squareup.picasso.Picasso
+
+
+class FragmentMoviesDetails : Fragment() {
+    private var actorsRecyclerView: RecyclerView? = null
+    private var movieId: Int? = null
+
+    private val actorsAdapter = ActorsAdapter()
+
+    private val moviesDetailsViewModel: MoviesDetailsViewModel by viewModels { MoviesDetailsViewModelFactory() }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
+            movieId = it.getInt(PARAM_ID)
+        }
+    }
+
+    override fun onCreateView(
+            inflater: LayoutInflater,
+            container: ViewGroup?,
+            savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        val view = inflater.inflate(R.layout.fragment_movies_details, container, false)
+
+        val backdropView = view.findViewById<ImageView>(R.id.backdrop_details)
+        val ageCategoryView = view.findViewById<TextView>(R.id.age_category)
+        val titleView = view.findViewById<TextView>(R.id.title_details)
+        val genresView = view.findViewById<TextView>(R.id.genres_details)
+        val ratingView = view.findViewById<RatingBar>(R.id.movie_rating_details)
+        val reviewView = view.findViewById<TextView>(R.id.reviews_details)
+        val descriptionView = view.findViewById<TextView>(R.id.description_details)
+        val castView = view.findViewById<TextView>(R.id.cast)
+        actorsRecyclerView = view.findViewById(R.id.actors_recycler_view)
+
+        movieId?.let { moviesDetailsViewModel.getMovieFromList(it) }
+
+
+        moviesDetailsViewModel.movieLiveData.observe(this.viewLifecycleOwner) { selectedMovie ->
+            selectedMovie?.let { movie ->
+                Picasso.get().load(Uri.parse(movie.backdrop)).into(backdropView)
+
+                ageCategoryView.text = context?.getString(R.string.age_category, movie.minimumAge)
+                titleView.text = movie.title
+                genresView.text = movie.genres.joinToString { it.name }
+                ratingView.rating = movie.ratings / 2
+                reviewView.text = context?.getString(R.string.reviews, movie.numberOfRatings)
+                descriptionView.text = movie.overview
+
+
+                val actors = movie.actors
+
+                if (actors.isEmpty()) {
+                    castView.visibility = View.GONE
+                } else {
+                    setUpActorsRecyclerView()
```

Comment:
Again, as for me setup of view is different from data binding

Reply:
do you mean put data binding into a separate function?